hoomd/hpmc/UpdaterMuVT.h

```diff
@@ -1650,6 +1650,28 @@ bool UpdaterMuVT<Shape>::tryRemoveParticle(uint64_t timestep, unsigned int tag,
         // do we have to compute energetic contribution?
         auto patch = m_mc->getPatchEnergy();
 
+     	// do we have to compute a wall contribution?
+    	auto field = m_mc->getExternalField();
+
+     	if (field)
+     	    {
+     	   // getPosition() takes into account grid shift, correct for that
+            Scalar3 p = m_pdata->getPosition(tag) + m_pdata->getOrigin();
+     	    int3 tmp = make_int3(0, 0, 0);
+     	    m_pdata->getGlobalBox().wrap(p, tmp);
+     	    vec3<Scalar> pos(p);
+     	    const BoxDim box = this->m_pdata->getGlobalBox();
+     	    unsigned int type = this->m_pdata->getType(tag);
+     	    quat<Scalar> orientation(m_pdata->getOrientation(tag));
+     	    lnboltzmann += field->energy(box,
+     	                                 type,
+     	                                 pos,
+     	                                 quat<float>(orientation),
+     	                                 1.0, // diameter i
+     	                                 0.0 // charge i
```

Comment:
We should get the diameter and charge from the particle data rather than assuming their values.

Reply:
The inserted particle does not exist in the particle data. If you have a need to insert particles with non-default charges, feel free to make a PR that adds the inserted particle charge as a parameter. `diameter` is unused.

Reply:
This is inside `tryRemoveParticle` so the diameter and charge do exist in the particle data.

Reply:
agreed